src/WebJobs.Extensions.DurableTask/EndToEndTraceHelper.cs

```diff
@@ -54,6 +59,41 @@ public static string LocalSlotName
 
 #pragma warning disable SA1117 // Parameters should be on same line or separate lines
 
+        private void SanitizeString(string? rawPayload, out string iloggerString, out string durableKustoTableString, bool isReplay = false)
+        {
+            if (isReplay)
+            {
+                iloggerString = rawPayload ?? "(replay)";
```

Comment:
I see `iloggerString` assigned here, but it's reassigned on line 74 (if `isReplay` is true). Seems like this line is unnecessary? Let me know if I'm misunderstanding this.

Reply:
I think this behavior was buggy, thanks for calling this out.
To be honest - part of why this method reads to strange is that current own logging is pretty inconsistent. **Sometimes** when the orchestrator is emitting a replay log, it replaces the inputs/outputs/exceptions with "(replay)", but sometimes it replaces it with "(replayed"), and sometimes it doesn't replace it at all.

I'm _ok_ making a small breaking change here to simplify this behavior. In the latest version of this PR, I completely drop the replacing of data with "(replay)" when "isReplay" is true. That simplifies the sanitizer's behavior and makes the logging more consistent across the board.

Reply:
Thanks for the explanation! I agree that we should make the behavior consistent across replays, but I'm not sure that we should remove this information. Do you know what the best practices are to make breaking changes with telemetry? Should we just add `(replay)` and then add the payload in the same string so we're not removing information that customers might use? I'm open to thoughts about this. I just want to get a better understanding.

Reply:
Resolving this comment after syncing with David offline about this.